gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/ConfigurableCleanableDataset.java

```diff
@@ -96,6 +106,14 @@
   private final List<VersionFinderAndPolicy<T>> versionFindersAndPolicies;
 
   /**
+   * A set of all available {@link RetentionActionFactory}s
+   */
+  private static final Set<Class<? extends RetentionActionFactory>> RETENTION_ACTION_TYPES;
```

Comment:
It lists out all the available `RetentionActionFactory`s. It gets updated when we add a new type of `RetentionAction`. 
It could be auto discoverable through reflections or service loader. We can add the auto discover feature when we need to support user supplied `RetentionAction`